src/shared/components/case-editor/case-edit-page/case-edit-page.component.ts

```diff
@@ -123,8 +126,46 @@ export class CaseEditPageComponent implements OnInit, AfterViewChecked {
     this.setFocusToTop();
   }
 
+  // Adding validation message to show it as Error Summary
+  generateErrorMessage(fields: CaseField[]){   
+    fields.filter(casefield => !this.caseFieldService.isReadOnly(casefield))
+          .forEach(casefield => {
+            if((this.editForm.controls['data'].get(casefield.id).hasError('required'))){
```

Comment:
Instead of referring to `this.editForm.controls['data'].get(casefield.id)` multiple times, you could do `const caseField = this.editForm.controls['data'].get(casefield.id);` at the top of this section and use the variable instead. It would make code more readable and would probably increase performance.

Reply:
As mentioned used variable to avoid repetitive code.